mongo/assets/configuration/spec.yaml

```diff
@@ -126,6 +126,43 @@ files:
       description: Path to the ca_certs file.
       value:
         type: string
+    - name: ssl_match_hostname
+      description: If `False`, disables SSL hostname verification.
+      value:
+        type: boolean
+        example: true
+    - name: tls
+      description: If `True`, create the connection to the server using transport layer security.
+      value:
+        type: boolean
+        example: false
+    - name: tlsCertificateKeyFile
+      description: A file containing the client certificate and private key. Implies tls=True.
+      value:
+        type: string
+    - name: tlsCAFile
+      description: |
+        A file containing one or a bundle of “certification authority” certificates, 
+        which are used to validate certificates passed from the other end of the connection. 
+        Implies tls=True.
+      value:
+        type: string
+    - name: tlsAllowInvalidHostnames
+      description: |
+        If True, disables TLS hostname verification. tlsAllowInvalidHostnames=False implies tls=True. 
+        Think very carefully before setting this to True as that could make your application 
+        vulnerable to on-path attackers.
+      value:
+        type: boolean
+        example: false
+    - name: tlsAllowInvalidCertificates
```

Comment:
We are mixing camel case and snake case for the names of the attributes. Do we have a standard?

Reply:
Good point.
This is a special case, as here we put the ssl (or tls in later versions) parameters that will be used to create the client to mongodb. The syntax is exactly the same as used in pymongo:
[4.2.0](https://pymongo.readthedocs.io/en/stable/api/pymongo/mongo_client.html)
[3.2.1](https://api.mongodb.com/python/3.2.1/api/pymongo/mongo_client.html)
The problem that you very well highlight is that they have also made that change in nomenclature.
In this case I would prefer to keep our config variable names the same as the 'standard' mongodb ones, ie `tlsCertificateKeyFile` instead of `tls_certificate_key_file`. I think that changing them could lead the user to errors. I don't know what @hithwen will think about it...

Reply:
We have other examples of integrations exposing library parameters with a different name so it should not be a problem, however in this case this params are not only from the library but can be used on a connection string too which I guess is why they've named them this way even for the python library. I think in this case it makes sense to leave the way mongo documents them to be explicit in that they are exactly that and the same